crates/runtime/src/instance.rs

```diff
@@ -104,6 +108,45 @@ pub trait ResourceLimiter {
     }
 }
 
+impl ResourceLimiter for Option<&mut dyn ResourceLimiter> {
```

Comment:
This I'm a bit wary of but only because this is part of the public API of the `wasmtime` crate (since `ResourceLimiter` is reexported in `wasmtime`). This honestly seems kinda ok for an implementation though if anyone else really wants to use it, all I'd say that that it should switch to `Option<T: ResourceLimiter>` instead of having `&mut dyn` here

Reply:
I decided against this and just defined two auxiliary functions in memory.rs instead.